cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2392,6 +2392,29 @@ struct NonZeroVisitor {
 
     return Status::OK();
   }
+
+  template <typename Type>
+  enable_if_t<is_decimal128_type<Type>::value || is_decimal256_type<Type>::value, Status>
```

Comment:
Can use `is_decimal_type` for more future-proof code.

Reply:
done